lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -37,7 +37,7 @@ class ProxyGenerator
      * Used to match very simple id methods that don't need
      * to be decorated since the identifier is known.
      */
-    const PATTERN_MATCH_ID_METHOD = '((public\s+)?(function\s+%s\s*\(\)\s*)\s*{\s*return\s*\$this->%s;\s*})i';
+    const PATTERN_MATCH_ID_METHOD = '((public\s+)?(function\s+%s\s*\(\)\s*)\s*(?::\s*[\w\\\\]+\s*)?{\s*return\s*\$this->%s;\s*})i';
```

Comment:
@Ocramius But there can be more identifiers separated by backslash - namespaces. This version is simple, I could extend it to include `\x7f-\xff` or use strict version like `(?::\s*\\?[a-z_\x7f-\xff][\w\x7f-\xff]*(?:\\[a-z_\x7f-\xff][\w\x7f-\xff]*)*\s*)`. But is strict really necessary? The checked code have to be valid PHP anyway and there cannot be any code between closing `)` and opening `{`

I'll try to prepare test, is there any existing testcase for this feature? I didn't find any...